rust/arrow/src/array.rs

```diff
@@ -609,8 +609,8 @@ impl From<ArrayDataRef> for BinaryArray {
 
 impl<'a> From<Vec<&'a str>> for BinaryArray {
     fn from(v: Vec<&'a str>) -> Self {
-        let mut offsets = vec![];
-        let mut values = vec![];
+        let mut offsets = Vec::with_capacity(v.len());;
```

Comment:
Sorry, I guess this should be `Vec::with_capacity(v.len() + 1)`. Also you have an extra `;`

Reply:
Thanks!